### ISSUE TITLE
[FR] Support custom map styles

### ISSUE THREAD

**Issue by Misalf-git** [organicmaps/organicmaps#1145]
Would it be possible to allow the user to change the map style to a custom made one?

Appart from changing map colors to personal preference, this might also allow highlighting things interesting to the user in colors that pop out (like hiking paths, bicycle ways) or even increasing map feature visibility (POI zoom levels) or removing unwanted map features to increase rendering speed.

I find these colors quite pleasant to look at and important roads are nicely visible (app is Cruiser but might as well be the default Mapsforge map style, idk):
![Screenshot_20210821-155206_Cruiser.png](https://user-images.githubusercontent.com/72605561/130324952-d7c7f381-e719-4725-b7ac-fd2b91dca549.png)

The default map style is somewhat too pale for me in order to easily distinguish certain features on the map.

**Comment by biodranik** [comment]
Editing map styles is not an easy experience. Adding this feature in the app looks now as overkill. However, users may create/edit styles on their computers, and then we may implement a simple way to replace them. Our styles are in data/styles folder.

The better idea is to collect users feedback, and create several focused styles embedded in the map and easily switchable:
1. Car navigation style
2. Hike with trails highlighted
3. Bike style
4. City tour style
5. … something else useful

**Comment by Misalf-git** [comment]
> Editing map styles is not an easy experience. Adding this feature in the app looks now as overkill.

I didn't thought of an in-app style editor but a text file containing map style configuration that could be loaded from the device storage. I understand if certain value changes via config file may not be feasible to be implemented. Changing the colors would already be very nice though.

**Comment by pastk** [comment]
> > Editing map styles is not an easy experience. Adding this feature in the app looks now as overkill.
> 
> I didn't thought of an in-app style editor but a text file containing map style configuration that could be loaded from the device storage. I understand if certain value changes via config file may not be feasible to be implemented. Changing the colors would already be very nice though.

Changing colors only is a much easier task, following is the config file for them:

https://github.com/organicmaps/organicmaps/blob/c9c9ad48b949da7a7ce609354518f7db818bc52b/data/styles/clear/style-clear/colors.mapcss#L59-L69

But to apply the changes ATM one needs to run the `tools/unix/generate_drules.sh` script and then re-build the app.

**Comment by biodranik** [comment]
Rebuilding the app is not necessary. You need to put regenerated style files into the writable folder of the app and relaunch the app.

**Comment by Nic787** [comment]
I agree with the fact that OM need to modernize or improve the color scheme in the app. Right now the colors are based on different shade of grey, orange and light green/yellow.

However, I think darker colors are more pleasant to the eyes and make it easier to differentiate some areas. OP example is great showing darker green and you can really see what’s a forest, what are the limit of the city and what’s farmland/grass field.

**Comment by omentic** [comment]
Where is the writable folder for map styles located on Android?

**Comment by pastk** [comment]
> Where is the writable folder for map styles located on Android?

On my device its
```
storage/emulated/0/Android/data/app.organicmaps/files/styles/
```

**Comment by FireFragment** [comment]
> On my device its
> ```
> storage/emulated/0/Android/data/app.organicmaps/files/styles/
> ```

I don't have the `styles` folder there... I would like to try using my own style. Should I create the folder?

![Screenshot_20240813-120245.png](https://github.com/user-attachments/assets/6c2cd979-7aef-456b-b3ed-3a0cf5d95271)

**Comment by wiesehahn** [comment]
I would love to see some kind of OM style library, where styles by the community or organisations are gathered, which could then be loaded dynamically from within the app. :)

**Comment by hvtim** [comment]
> I would love to see some kind of OM style library, where styles by the community or organisations are gathered, which could then be loaded dynamically from within the app. :)

+1 to this. Using the ios app, and it's better than almost every way compared to apple maps. But the default dark mode orange/brown/red colors strain my eyes. It would be great if there were options to choose other community-made color profiles, or an option to disable building numbers to make the map more readable/less cluttered when zoomed in.
